### PR TITLE
STOR-1500: Switch to the new operator

### PR DESCRIPTION
**Pull request by jsafrane**
Build the aws-ebs-csi-driver-operator from the refactored sources.

This finishes https://github.com/openshift/enhancements/blob/master/enhancements/storage/csi-driver-operator-merge.md 
cc @openshift/storage 

~WIP: I added a second commit with Dockerfile.rhel7, [something](https://github.com/openshift/release/pull/45622/files#diff-d3c90828d0b4a3954ff9498d3c2196b600508244ece93a922fd3f8df79cc38fdR10) is replacing reference to Dockerfile.aws-ebs in CI config.~
CI is fine with Dockerfile.aws-ebs now.

### CODE REVIEW

**Comment by jsafrane** [comment]
/hold
this repo does not wait for qe-approved (yet), holding for QE manually.

**Comment by jsafrane** [comment]
/test e2e-aws-csi
With https://github.com/openshift/csi-operator/pull/70, there should be `"Running openshift/csi-operator for ebs.csi.aws.com"` in the logs

File: Dockerfile.rhel7 [hunk 1]
```diff
@@ -0,0 +1 @@
+Dockerfile.aws-ebs
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply Dockerfile.rhel7:1]: This symlink is here to fix buids in CI, [something](https://redhat-internal.slack.com/archives/CBN38N3MW/p1700142212251799?thread_ts=1700069186.191119&cid=CBN38N3MW) is constantly [replacing](https://github.com/openshift/release/commit/aa3426c4664d7e1231252092df4ab6b7e7bbceb9#diff-d3c90828d0b4a3954ff9498d3c2196b600508244ece93a922fd3f8df79cc38fdR10) `dockerfile_path: Dockerfile.aws-ebs` with `Dockerfile.rhel7` there.

[the remark this answers is not in the document]
Comment (jsafrane) [reply Dockerfile.rhel7:1]: And the symlink needs to be removed before we merge this PR, that's why it's WIP.

**Comment by jsafrane** [comment]
/retest
`storage` ClusterOperator looks OK (Available=true, Progressing=false) in the last hypershift-aws-e2e-external run.

**Comment by jsafrane** [comment]
/test hypershift-aws-e2e-external

Cluster upgrade failed, but it looks like a flake

File: Dockerfile.rhel7 (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsafrane) [reply Dockerfile.rhel7]: CI is fine now, Dockerfile.rhel7 is removed.

**Comment by jsafrane** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.15-e2e-aws-ovn-upi

**Comment by jsafrane** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.15-e2e-aws-ovn-upgrade

**Comment by jsafrane** [comment]
/payload-abort

some jobs failed with `duplicate image name 'aws-ebs-csi-driver-operator-test'`. Trying to fix it in https://github.com/openshift/release/pull/46067

**Comment by jsafrane** [comment]
I found [a thread](https://redhat-internal.slack.com/archives/CBN38N3MW/p1690558682493439) about this, in the end:

> You have an image (name) which is both built and imported, that's invalid.

A payload test cannot run on a repo that builds [a test image](https://github.com/openshift/release/blob/ec6c81d42239b45924fd0700e5b715e2205ecf40/ci-operator/config/openshift/csi-operator/openshift-csi-operator-master.yaml#L27-L29) that's then imported by [the payload job](https://github.com/openshift/release/blob/ec6c81d42239b45924fd0700e5b715e2205ecf40/ci-operator/config/openshift/release/openshift-release-master__nightly-4.15.yaml#L34-L37).

I don't understand why some tests actually did run, but re-running those that can succeed:

/payload 4.15 nightly blocking
/payload 4.15 ci blocking

**Comment by ropatil010** [comment]
Premerge testing Pass. 
Payload: 4.15.0-0.test-2023-11-22-044250-ci-ln-2tg7itk-latest 

Tested on 5 profiles and added results here: [STOR-1543](https://issues.redhat.com/browse/STOR-1543 )
1. aos-4_15/ipi-on-aws/versioned-installer-customer_vpc-http_proxy-sts
2. aos-4_15/ipi-on-aws/versioned-installer-customer_vpc-disconnected_private_cluster
3. aos-4_15/ipi-on-aws/versioned-installer-fips-ci
4. aos-4_15/ipi-on-aws/versioned-installer-ovn-hypershift-ci
5. aos-4_15/ipi-on-aws/versioned-installer-kms-etcd_encryption-ci

CI result looks good. 
/label qe-approved

**Comment by jsafrane** [comment]
rebased

**Comment by jsafrane** [comment]
/hold cancel
QE approved, CI is green, payload jobs were ok-ish yesterday

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Pull request merged**